src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -245,12 +245,20 @@ int dtrace_waited_probe(ObjectMonitor* monitor, Handle obj, JavaThread* thr) {
   return 0;
 }
 
-static const int NINFLATIONLOCKS = 256;
-static PlatformMutex* gInflationLocks[NINFLATIONLOCKS];
+static constexpr size_t inflation_lock_count() {
+  return 256;
+}
+
+// Static storage for an array of PlatformMutex.
+alignas(PlatformMutex) static uint8_t _inflation_locks[inflation_lock_count()][sizeof(PlatformMutex)];
```

Comment:
Sorry I am still not understanding why this can't just be:
```
alignas(PlatformMutex) static PlatformMutex _inflation_locks[inflation_lock_count()];
```
??? (And is the alignas even needed?)

Reply:
`alignas` is not needed in your example. 

I guess we do not do this because we want to control the initialization of these objects and not rely on C++ dynamic initialization. 

You could solve this in different ways. Some containing struct, which was rejected in the #13143. Or rewriting PlatformMutex to be constant initializable. 

Having a byte blob as backing storage does seem like the simplest least intrusive solution.

When we move to C++17 then this could be a candidate use for `std::optional`.

Reply:
`alignas(PlatformMutex) static PlatformMutex _inflation_locks[inflation_lock_count()];` will not work as the compiler will use default initialization for the array, causing the constructor of PlatformMutex to be called during library loading by the dynamic linker.

With the current `uint8_t` usage, we have to ensure the compiler places the array at a suitable aligned boundary for `PlatformMutex` otherwise its free to place it at `alignas(uint8_t)` AFAIK, which can be 1.

Reply:
Ugghh sorry not thinking in C++ - forgot about the implicit initialization in that case.

Reply:
That's okay, so did I originally.